src/dguweb/web/router.ex

```diff
@@ -45,7 +46,10 @@ defmodule DGUWeb.Router do
   end
 
   # Other scopes may use custom stacks.
-  # scope "/api", DGUWeb do
-  #   pipe_through :api
-  # end
+  scope "/api", DGUWeb do
+     pipe_through :api
+
+     get "/autocomplete", AutocompleteController, :query
+
```

Comment:
How about using the same URL as search, ie `/search?q=abc` and if the Accept header is application/json you return the JSON version of the results?